abi-checker needs `-fPIC` for libraries that use Qt

The abi-checker currently emits the following error when running on a library that uses Qt (eg. https://build.osrfoundation.org/job/ignition_gazebo-abichecker-any_to_any-ubuntu_auto-amd64/6364/ for https://github.com/gazebosim/gz-sim/pull/1989)

```

The GCC parameters:
  gcc -fdump-lang-raw -fkeep-inline-functions -c -x c++ -fpermissive -w  -std=c++17 "/tmp/of3phyEMhD/dump1.h"  -I/usr/include/OGRE-2.2/Hlms/Common -I/usr/local/destination_branch/include/ignition/gazebo6 -I/usr/include/eigen3 -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/ignition/utils1 -I/usr/include/ignition/transport11 -I/usr/include/ignition/sdformat12 -I/usr/include/ignition/rendering6 -I/usr/include/ignition/plugin1 -I/usr/include/ignition/physics5 -I/usr/include/ignition/msgs8 -I/usr/include/ignition/math6 -I/usr/include/ignition/gui6 -I/usr/include/ignition/common4 -I/usr/include/ignition/cmake2 -I/usr/include/x86_64-linux-gnu/qt5/QtWidgets -I/usr/include/x86_64-linux-gnu/qt5/QtQuickControls2 -I/usr/include/x86_64-linux-gnu/qt5/QtQuick -I/usr/include/x86_64-linux-gnu/qt5/QtQml -I/usr/include/x86_64-linux-gnu/qt5/QtGui -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I/usr/include/OGRE/RenderSystems/GLES2

In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/QtCore:4,
                 from /usr/include/ignition/gui6/gz/gui/qt.h:25,
                 from /usr/include/ignition/gui6/gz/gui/Application.hh:25,
                 from /usr/include/ignition/gui6/ignition/gui/Application.hh:18,
                 from /usr/local/destination_branch/include/ignition/gazebo6/ignition/gazebo/gui/Gui.hh:23,
                 from /usr/local/destination_branch/include/ignition/gazebo6/ignition/gazebo.hh:139,
                 from /tmp/of3phyEMhD/dump1.h:3:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qglobal.h:1187:4: error: #error "You must build your code with position independent code if Qt was built with -reduce-relocations. " "Compile your code with -fPIC (-fPIE is not enough)."
 1187 | #  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
      |    ^~~~~
```

It then reports
```
ERROR: some errors occurred when compiling headers
ERROR: see log for details:
  /home/jenkins/workspace/ignition_gazebo-abichecker-any_to_any-ubuntu_auto-amd64/abi_checker/logs/ign-gazebo/X/log.txt
```

in the Jenkins page. It's not clear what that the error actually causes since the compatibility report is still generated, but I wonder if we're getting incomplete results because of this.

--- a/jenkins-scripts/docker/ignition-abichecker.bash
+++ b/jenkins-scripts/docker/ignition-abichecker.bash
@@ -92,7 +92,7 @@ then
   fi
 fi
 
-if [[ "${ABI_JOB_SOFTWARE_NAME}" = "ign-gazebo"   && ${GZ_NAME_PREFIX_MAJOR_VERSION} -ge 7 ]] || \
+if [[ "${ABI_JOB_SOFTWARE_NAME}" = "ign-gazebo"   && ${GZ_NAME_PREFIX_MAJOR_VERSION} -ge 6 ]] || \
   [[ "${ABI_JOB_SOFTWARE_NAME}" = "ign-gui"       && ${GZ_NAME_PREFIX_MAJOR_VERSION} -ge 7 ]] || \
   [[ "${ABI_JOB_SOFTWARE_NAME}" = "ign-launch"    && ${GZ_NAME_PREFIX_MAJOR_VERSION} -ge 6 ]] || \
   [[ "${ABI_JOB_SOFTWARE_NAME}" = "ign-sensors"   && ${GZ_NAME_PREFIX_MAJOR_VERSION} -ge 7 ]] || \

